server/src/main/java/org/elasticsearch/index/mapper/vectors/TokenPruningConfig.java

```diff
@@ -163,16 +171,120 @@ public static TokenPruningConfig fromXContent(XContentParser parser) throws IOEx
                 } else {
                     throw new ParsingException(
                         parser.getTokenLocation(),
-                        "[" + PRUNING_CONFIG.getPreferredName() + "] does not support [" + currentFieldName + "]"
+                        "[" + PRUNING_CONFIG_FIELD + "] does not support [" + currentFieldName + "]"
                     );
                 }
             } else {
                 throw new ParsingException(
                     parser.getTokenLocation(),
-                    "[" + PRUNING_CONFIG.getPreferredName() + "] unknown token [" + token + "] after [" + currentFieldName + "]"
+                    "[" + PRUNING_CONFIG_FIELD + "] unknown token [" + token + "] after [" + currentFieldName + "]"
                 );
             }
         }
         return new TokenPruningConfig(ratioThreshold, weightThreshold, onlyScorePrunedTokens);
     }
+
+    public static TokenPruningConfig parseFromMap(Map<String, Object> pruningConfigMap) {
+        Object mappedTokensFreqRatioThreshold = pruningConfigMap.remove(TOKENS_FREQ_RATIO_THRESHOLD.getPreferredName());
+        Object mappedTokensWeightThreshold = pruningConfigMap.remove(TOKENS_WEIGHT_THRESHOLD.getPreferredName());
+        Object mappedOnlyScorePrunedTokens = pruningConfigMap.remove(ONLY_SCORE_PRUNED_TOKENS_FIELD.getPreferredName());
+
+        if (pruningConfigMap.isEmpty() == false) {
+            throw new MapperParsingException("[" + PRUNING_CONFIG_FIELD + "] has unknown fields");
+        }
+
+        Float tokensFreqRatioThreshold = parseTokensFreqRatioThreshold(mappedTokensFreqRatioThreshold);
+        Float tokensWeightThreshold = parseTokensWeightThreshold(mappedTokensWeightThreshold);
+        boolean onlyScorePrunedTokens = mappedOnlyScorePrunedTokens != null ? parseScorePrunedTokens(mappedOnlyScorePrunedTokens) : false;
+
+        if (tokensFreqRatioThreshold != null || tokensWeightThreshold != null) {
+            return new TokenPruningConfig(tokensFreqRatioThreshold, tokensWeightThreshold, onlyScorePrunedTokens);
+        }
+
+        return null;
+    }
+
+    private static Float parseFloatNumberFromObject(Object numberObject) {
+        if (numberObject instanceof Integer intValue) {
+            return (float) intValue;
+        } else if (numberObject instanceof Float floatValue) {
+            return floatValue;
+        } else if (numberObject instanceof Double doubleValue) {
+            return ((Double) numberObject).floatValue();
+        }
+        return null;
```

Comment:
Should this throw if it's not a number instead of doing multiple validation checks on null?